projects/core/components/calendar/calendar.component.ts

```diff
@@ -136,4 +148,13 @@ export class TuiCalendarComponent implements TuiWithOptionalMinMax<TuiDay> {
         this.hoveredItem = day;
         this.hoveredItemChange.emit(day);
     }
+
+    private getTuiMonth(value: TuiDay | TuiDayRange | null): TuiMonth | null {
+        const possibleTuiDay = value as TuiDay;
```

Comment:
You can do `instanceof` test since `TuiDay` is a class.

Reply:
fixed